src/mito2/src/sst/parquet/row_group.rs

```diff
@@ -78,13 +84,14 @@ impl<'a> InMemoryRowGroup<'a> {
             row_group_idx,
             cache_manager,
             column_cached_pages: vec![None; metadata.columns().len()],
+            file_path,
+            object_store,
         }
     }
 
     /// Fetches the necessary column data into memory
-    pub async fn fetch<T: AsyncFileReader + Send>(
+    pub async fn fetch(
```

Comment:
I wonder if you would be interested in doing some refactoring. I think we can add a structure called `RowGroupsFetcher`, and its `fetch` method will produce an `InMemoryRowGroup` that implements the `RowGroups` trait.

Reply:
I prefer a `PageFetcher` for fetching pages. It doesn't need to care about who uses the page it fetches.